src/modules/mc_rate_control/MulticopterRateControl.cpp

```diff
@@ -153,8 +153,14 @@ MulticopterRateControl::Run()
 			landing_gear_s landing_gear;
 
 			if (_landing_gear_sub.copy(&landing_gear)) {
-				if (landing_gear.landing_gear != landing_gear_s::GEAR_KEEP) {
-					_landing_gear = landing_gear.landing_gear;
+				if (landing_gear.landing_gear != landing_gear_s::GEAR_KEEP &&
+				    _v_control_mode.flag_control_manual_enabled) {
```

Comment:
I think I agree but here I was just trying to preserve how it was:
https://github.com/PX4/PX4-Autopilot/blob/dddd2c3297bd57a92d3b0261e2702f4f36d8a774/src/modules/commander/Commander.cpp#L2309

Reply:
ok, but it's different now. Trying to simplify it was:
```
Commander: if (manual_mode && landed) { publish gear from switch }
RateControl: actuation = subscribed gear
```
and now it is:
```
ManualControl: publish gear switch
RateControl: if (manual_mode && landed) { actuation = subscribed gear }
```

As a result if I understood correctly:
- any autonomous mode cannot move the gear anymore because its publication gets ignored
- in autonomous modes you can now move the gear by switch (even when landed)

Reply:
I suggest to enable the gear always to be changed unless we're landed and want to put it up.